tests/PHPStan/Rules/PhpDoc/IncompatibleRequireExtendsTypeRuleTest.php

```diff
@@ -0,0 +1,62 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\PhpDoc;
+
+use PHPStan\Rules\ClassCaseSensitivityCheck;
+use PHPStan\Rules\Rule;
+use PHPStan\Testing\RuleTestCase;
+
+/**
+ * @extends RuleTestCase<IncompatibleRequireExtendsTypeRule>
+ */
+class IncompatibleRequireExtendsTypeRuleTest extends RuleTestCase
+{
+
+	protected function getRule(): Rule
+	{
+		$reflectionProvider = $this->createReflectionProvider();
+
+		return new IncompatibleRequireExtendsTypeRule(
+			$reflectionProvider,
+			new ClassCaseSensitivityCheck($reflectionProvider, true),
+			new UnresolvableTypeHelper(),
+			true,
+		);
+	}
+
+	public function testRule(): void
+	{
+		$this->analyse([__DIR__ . '/data/incompatible-require-extends.php'], [
+			[
+				'PHPDoc tag @require-extends contains invalid type IncompatibleRequireExtends\SomeTrait.',
+				8,
+			],
+			[
+				'PHPDoc tag @require-extends contains invalid type IncompatibleRequireExtends\SomeInterface.',
+				13,
+			],
+			[
+				'PHPDoc tag @require-extends contains invalid type IncompatibleRequireExtends\SomeEnum.',
+				18,
+			],
+			[
+				'PHPDoc tag @require-extends contains unknown class IncompatibleRequireExtends\TypeDoesNotExist.',
+				23,
+				'Learn more at https://phpstan.org/user-guide/discovering-symbols',
+			],
+			[
+				'PHPDoc tag @require-extends cannot contain generic type.',
```

Comment:
this error messge is a bit misleading. my idea was that in `@phpstan-require-extends SomeClass` we only name the class by its name, without generics params. 

that way we don't need to validate the generics and therefore safe phpstan from validating them twice.

.. I think we just need a better error message.. wdyt?